tests/acceptance/bootstrap/GraphContext.php

```diff
@@ -2958,6 +2958,36 @@ public function userListsTheActivitiesForFolderOfSpaceWithDepthOrLimitUsingTheGr
 		$this->featureContext->setResponse($response);
 	}
 
+	/**
+	 * @Then the activities should be in the following order:
+	 *
+	 * @param TableNode $table
+	 *
+	 * @return void
+	 * @throws Exception
+	 */
+	public function theActivitiesShouldBeInTheFollowingOrder(TableNode $table): void {
+		$responseBody = $this->featureContext->getJsonDecodedResponseBodyContent();
+		$activities = $responseBody->value;
+		$expectedOrder = [];
+		foreach ($table->getHash() as $row) {
+			$expectedOrder[] = [
+				'resource' => $row['resource'],
+				'message' => $row['message']
+			];
+		}
+
+		foreach ($expectedOrder as $index => $expectedActivity) {
+			$actualActivity = $activities[$index];
+			if ($actualActivity->template->variables->resource->name !== $expectedActivity['resource']) {
+				throw new Exception("Expected resource '{$expectedActivity['resource']}' but got '{$actualActivity->template->variables->resource->name}' at index {$index}.");
+			}
+			if ($actualActivity->template->message !== $expectedActivity['message']) {
+				throw new Exception("Expected message '{$expectedActivity['message']}' but got '{$actualActivity->template->message}' at index {$index}.");
+			}
```

Comment:
maybe build a string and compare them?
```suggestion
$expectedActivity = "$property: $expectedValue";
$actualActivity = $actualActivity->template->variables->resource->name . ":" . $actualActivity->template->message;
Assert::assertEquals($expectedActivity, $actualActivity, "some error message");
```